smartsim/_core/_install/builder.py

```diff
@@ -756,30 +795,20 @@ def _extract_download(
             zip_file.extractall(target)
 
 
-@t.final
 @dataclass(frozen=True)
 class _PTArchive(_WebZip, _RAIBuildDependency):
-    os_: OperatingSystem
+    architecture: Architecture
     device: TDeviceStr
     version: str
 
-    @property
-    def url(self) -> str:
-        if self.os_ == OperatingSystem.LINUX:
-            if self.device == "gpu":
-                pt_build = "cu117"
-            else:
-                pt_build = "cpu"
-            # pylint: disable-next=line-too-long
-            libtorch_arch = f"libtorch-cxx11-abi-shared-without-deps-{self.version}%2B{pt_build}.zip"
-        elif self.os_ == OperatingSystem.DARWIN:
-            if self.device == "gpu":
-                raise BuildError("RedisAI does not currently support GPU on Macos")
-            pt_build = "cpu"
-            libtorch_arch = f"libtorch-macos-{self.version}.zip"
-        else:
-            raise BuildError(f"Unexpected OS for the PT Archive: {self.os_}")
-        return f"https://download.pytorch.org/libtorch/{pt_build}/{libtorch_arch}"
+    @staticmethod
+    def supported_platforms() -> t.Sequence[t.Tuple[OperatingSystem, Architecture]]:
+        # TODO: Rework this to do a recursive search
+        return tuple(
+            itertools.chain.from_iterable(
+                var.supported_platforms() for var in _PTArchive.__subclasses__()
+            )
+        )
```

Comment:
Not wrong just be warned: If someone subclass this class and forgets to re-implement this method, that will lead to a recursion error

Reply:
Yah that was the point of the above comment, but I'll make it a bit clearer

Reply:
Oh! Is that what that meant? I thought that was talking about if someone where to subclass `_PTArchiveLinux` or `_PTArchiveMacOSX`.


I meant something like
```py
calss _PTArchiveWindows(_PTArchive):
    # Very explicitly not overwriting `supported_platforms`
    ...
```
will now raise a
```
RecursionError: maximum recursion depth exceeded
```

Reply:
Ohh I see, yeah that makes sense.